sound/soc/codecs/rt1316-sdw.c

```diff
@@ -0,0 +1,708 @@
+	regmap_write(rt1316->regmap, 0xc60e, 0x05);
+	regmap_write(rt1316->regmap, 0xc00f, 0x01);
+	regmap_write(rt1316->regmap, 0xc09c, 0x7b);
+
+	regmap_write(rt1316->regmap, 0xc602, 0x07);
+	regmap_write(rt1316->regmap, 0xc603, 0x07);
+
+	regmap_write(rt1316->regmap, 0xc0a3, 0x71);
+	regmap_write(rt1316->regmap, 0xc00b, 0x20);
+
+	regmap_write(rt1316->regmap, 0xc093, 0x80);
+	regmap_write(rt1316->regmap, 0xc09d, 0x80);
+	regmap_write(rt1316->regmap, 0xc0b0, 0x77);
+	regmap_write(rt1316->regmap, 0xc050, 0x83);
+	regmap_write(rt1316->regmap, 0xc010, 0xa5);
+
+	regmap_write(rt1316->regmap, 0x2232, 0x80);
+	regmap_write(rt1316->regmap, 0xc0b0, 0x77);
+	regmap_write(rt1316->regmap, 0xc011, 0x00);
+	regmap_write(rt1316->regmap, 0xc020, 0x00);
+	regmap_write(rt1316->regmap, 0xc023, 0x00);
+	regmap_write(rt1316->regmap, 0x3101, 0x00);
+	regmap_write(rt1316->regmap, 0x3004, 0xd4);
+	regmap_write(rt1316->regmap, 0x3005, 0xf6);
+	regmap_write(rt1316->regmap, 0xc007, 0x44);
+	regmap_write(rt1316->regmap, 0xc008, 0x44);
+	regmap_write(rt1316->regmap, 0xc009, 0x33);
+	regmap_write(rt1316->regmap, 0xc007, 0x11);
+	regmap_write(rt1316->regmap, 0xc008, 0x11);
+	regmap_write(rt1316->regmap, 0xc009, 0x00);
+	regmap_write(rt1316->regmap, 0xc022, 0xd6);
+	regmap_write(rt1316->regmap, 0xc025, 0xd6);
+
+	regmap_write(rt1316->regmap,
+		SDW_SDCA_CTL(FUN_SMART_AMP, RT1316_SDCA_ENT_XU24, RT1316_SDCA_CTL_BYPASS, 0), 0x00);
+
+	regmap_write(rt1316->regmap, 0xd001, 0x03);
+	regmap_write(rt1316->regmap, 0xd002, 0xbf);
+	regmap_write(rt1316->regmap, 0xd003, 0x03);
+	regmap_write(rt1316->regmap, 0xd004, 0xbf);
+
+	if (rt1316->first_hw_init) {
+		regcache_cache_bypass(rt1316->regmap, false);
+		regcache_mark_dirty(rt1316->regmap);
+	} else
+		rt1316->first_hw_init = true;
+
+	/* Mark Slave initialization complete */
+	rt1316->hw_init = true;
+
+	pm_runtime_mark_last_busy(&slave->dev);
+	pm_runtime_put_autosuspend(&slave->dev);
+
+	dev_dbg(&slave->dev, "%s hw_init complete\n", __func__);
+	return 0;
+}
+
+static int rt1316_update_status(struct sdw_slave *slave,
+					enum sdw_slave_status status)
+{
+	struct  rt1316_sdw_priv *rt1316 = dev_get_drvdata(&slave->dev);
+
+	/* Update the status */
+	rt1316->status = status;
+
+	if (status == SDW_SLAVE_UNATTACHED)
+		rt1316->hw_init = false;
+
+	/*
+	 * Perform initialization only if slave status is present and
+	 * hw_init flag is false
+	 */
+	if (rt1316->hw_init || rt1316->status != SDW_SLAVE_ATTACHED)
+		return 0;
+
+	/* perform I/O transfers required for Slave initialization */
+	return rt1316_io_init(&slave->dev, slave);
+}
+
+static int rt1316_classd_event(struct snd_soc_dapm_widget *w,
+	struct snd_kcontrol *kcontrol, int event)
+{
+	struct snd_soc_component *component =
+		snd_soc_dapm_to_component(w->dapm);
+
+	switch (event) {
+	case SND_SOC_DAPM_POST_PMU:
+		snd_soc_component_write(component,
+			SDW_SDCA_CTL(FUN_SMART_AMP, RT1316_SDCA_ENT_PDE23, RT1316_SDCA_CTL_REQ_POWER_STATE, 0), PS0);
```

Comment:
in rt711_sdca_fu1e_event  it is done as the following. But here it is done with " snd_soc_component_write(component". please unify it, thanks.

````C
	switch (event) {
	case SND_SOC_DAPM_POST_PMU:
		regmap_write(rt711->regmap,
````

Reply:
right, don't you want a regmap write here?